lib/elastic_searcher.rb

```diff
@@ -14,6 +14,15 @@ def search
     @api ? result : [error_msg(e), result]
   end
 
+  def suggestions
+    result = Rubygem.__elasticsearch__.search(suggestions_definition).page(@page)
+    result = result.results.response.response[:suggest][:completion_suggestion][0][:options]
+    @api ? result.map(&:_source) : [nil, result]
+  rescue Faraday::ConnectionFailed, Faraday::TimeoutError, Elasticsearch::Transport::Transport::Error => e
+    result = Rubygem.legacy_search(@query).page(@page)
+    @api ? result : [error_msg(e), result]
```

Comment:
I think we need `@api` or double return for this function. Update `search` to pass `api` as an argument.
Also, move following from controller to `suggestions` function here, ie return array of gem names:
```
results.each do |gem|
      names << gem.name
    end
```

Reply:
Done this in commit `Move returning of gems name function from controller to model`. However, I may misunderstand the comment

> Update search to pass api as an argument.

Will you check the code, please?